ext/session/session.c

```diff
@@ -2711,6 +2723,22 @@ static const zend_function_entry session_functions[] = {
 
 /* {{{ SessionHandlerInterface functions[]
 */
+static const zend_function_entry php_session_save_handler_iface_functions[] = {
+	PHP_ABSTRACT_ME(SessionSaveHandlerInterface, open, arginfo_session_class_open)
```

Comment:
128 bits of entropy are a bit much for most DDoSs. Regardless of that, you could add it to the interface, and it would still be better than the old legacy API:

```php
interface SessionSaveHandlerInterface
{
    public function create() : SessionId;
    /** just save or refresh the damn thing - metadata can contain arbitrary info, such as TTL too */
    public function save(SessionId $id, SessionMetadata $metadata, SessionData $session) : void;
    public function destroy(SessionId $id) : void;
    public function load(SessionId $id) : SessionData;
}
```